nixos/modules/services/misc/ollama.nix

```diff
@@ -78,11 +78,11 @@ in
         '';
       };
       port = lib.mkOption {
-        type = types.nullOr types.ints.u16;
+        type = types.port;
```

Comment:
`services.ollama.port` was intentionally made nullable so `OLLAMA_HOST` could be set to just a host without a port, in case that's ever useful to someone. Why was this removed?

Also, this should be changed as well if you want to completely remove the code for null port:
``` diff
# line 135
environment = cfg.environmentVariables // {
   HOME = cfg.home;
   OLLAMA_MODELS = cfg.models;
-  OLLAMA_HOST =
-    if cfg.port == null then
-      cfg.host
-    else
-      "${cfg.host}:${toString cfg.port}";
+  OLLAMA_HOST = "${cfg.host}:${toString cfg.port}";
```

Reply:
Job done!

Reply:
but you didn't answer why nullability was removed

Reply:
Oh right, for two reasons:
1. Because IMHO, it doesn't make sense to me
2. There are very few occurrences of `types.nullOr types.port` in `nixos/services`, compared to `types.port`.

Reply:
you think it's unlikely that _anyone_ would _ever_ want to set `OLLAMA_HOST="127.0.0.1"`, without specifying a port?

Reply:
I believe yes. In that case, there's a default port which is set, so, this will never happen.

Reply:
If you prefer, I can revert that change, you're maintaining it, I don't want to hijack anything.

Reply:
nah, it's fine, I just made it like that when I split `listenAddress` because I was worried that _someone_ would eventually have some need to set just the host without the port, which it didn't support at first until I thought of making `port` nullable.

If you genuinely think it's unnecessary, I honestly would rather remove the added complexity it introduces.

Reply:
All good then :)